[bitnami/osclass] Add VIB tests to OsClass helm chart

mdhont opened the pull request:
Signed-off-by: Michiel <michield@vmware.com>


### Description of the change

Adding VIB tests to the OsClass Chart.

### Additional information

A Goss test for checking HTTPS connectivity on an alternative port (444) is [failing](https://github.com/mdhont/charts/runs/8304018374). This is outside of the scope of adding the tests but needs to be fixed.  I will create a separate task for it.

### Checklist

<!-- [Place an '[X]' (no spaces) in all applicable fields. Please remove unrelated fields.] -->

- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

joancafom commented on .vib/osclass/goss/goss.yaml:
```diff
@@ -0,0 +1,28 @@
+http:
+  http://osclass:{{ .Vars.service.ports.https }}:
```
```suggestion
  https://osclass:{{ .Vars.service.ports.https }}:
```

joancafom commented on .vib/osclass/goss/goss.yaml:
```diff
@@ -0,0 +1,28 @@
+    {{ range .Vars.hostAliases }}
+    {{ $ip := .ip }}
+    {{ range $host := .hostnames }}
+      - /{{ $ip }}.*{{$host}}/
```
```suggestion
      - /{{ $ip }}.*{{ $host }}/
```

joancafom reviewed (commented):
Thanks for your PR @mdhont ! I left some comments for you to review

FraPazGal commented on .vib/osclass/goss/goss.yaml:
```diff
@@ -0,0 +1,28 @@
+    exists: true
+    filetype: file
+    mode: "0666"
+    contains:
```
Could you please also check the correct DB credentials are used? Until now we usually didn't check them explicitly but discovered that the modified values could be parsed incorrectly by the chart and we would know because the app would also work with the default values.

mdhont replied:
As no credentials are used in the configuration file, I'm just checking the database name

joancafom replied:
I think you missed them, but it seems there are entries in the file with the configuration:

<img width="450" alt="Screenshot 2022-09-19 at 09 22 53" src="https://user-images.githubusercontent.com/24253844/190968004-1838dc88-b75c-426f-91b5-777b6592a4cb.png">

mdhont commented:
@FraPazGal, @joancafom, I've changed the following:

- Simplified selectors, URL's where possible
- Reverted to using the default SSL port as using a non default one breaks the application. I will create a separate task to fix this.
- Added a test for the database name, checking the credentials was not possible as they don't appear in the configuration file.
- Removed wrongly committed Moodle file.

joancafom commented on .vib/osclass/cypress/cypress/support/commands.js:
```diff
@@ -0,0 +1,24 @@
+Cypress.Commands.add(
+  'login',
+  (username = Cypress.env('username'), password = Cypress.env('password')) => {
+    cy.visit('oc-admin');
```
Use a consistent style towards URLs:

```suggestion
    cy.visit('/oc-admin');
```

joancafom commented on .vib/osclass/goss/goss.yaml:
```diff
@@ -0,0 +1,29 @@
+http:
+  https://osclass:{{ .Vars.service.ports.https }}:
+    status: 200
+    allow-insecure: true
```
I have checked the associated test in the description about port `444` and the related error in the [execution-graph](https://cp.bromelia.vmware.com/v1/execution-graphs/548980de-213e-4df5-902a-f31f7a448df8/tasks/b0506ae3-6dd3-4b18-9714-3aee51e26a52/result) is:

```json
{
"duration": 0.003,
"name": "HTTP http://osclass:444 status",
"location": "goss",
"message": "ERRORS: [HTTP: http://osclass:444: status: doesn't match, expect: [200] found: [400]] FAILURES: [HTTP: http://osclass:444: status: doesn't match, expect: [200] found: [400]] ",
"test_id": "4bff62fd",
"status": "FAILED"
},
```

The error is a 400 Bad Request, which does mean that GOSS was able to connect to that port and got a response. Hence, I don't think the actual change of port is failing. The error might be related to accessing a HTTPs port using a plain HTTP protocol. Could you please check it?

joancafom commented on .vib/osclass/goss/vars.yaml:
```diff
@@ -0,0 +1,18 @@
+  http: 8081
+service:
+  ports:
+    https: 443
```
You have updated the tests but not changed the port here

mdhont replied:
This keeps failing when changing the port to a non default one:
```
<testcase name="HTTP https://osclass:444 status" time="5.000">
<system-err>https://osclass:444: status: Error: Get https://osclass:444: net/http: request canceled (Client.Timeout exceeded while awaiting headers)</system-err>
<failure>https://osclass:444: status: Error: Get https://osclass:444: net/http: request canceled (Client.Timeout exceeded while awaiting headers)</failure>
</testcase>
```

mdhont replied:
I've added a timeout of 70000 and it works now

FraPazGal replied:
@mdhont Is such a huge timeout necessary? The almost 12 minutes used seems a bit overboard for me. Have you encountered instances where the tests got close to that length? From what I recall, local testing ranged between 20-60s for the test execution.

joancafom commented on .vib/osclass/vib-publish.json:
```diff
@@ -22,7 +22,7 @@
           "path": "/bitnami/osclass"
         },
-        "runtime_parameters": "Im9zY2xhc3NQYXNzd29yZCI6ICJiaXRuYW1pMSIKIm9zY2xhc3NVc2VybmFtZSI6ICJ1c2VyIgoic2VydmljZSI6CiAgInBvcnQiOiA4MAogICJ0eXBlIjogIkxvYWRCYWxhbmNlciI=",
+        "runtime_parameters": "b3NjbGFzc1VzZXJuYW1lOiB0ZXN0X3VzZXIKb3NjbGFzc1Bhc3N3b3JkOiBDb21wbGljYXRlZFBhc3N3b3JkMTIzITQKc2VydmljZToKICBwb3J0czoKICAgIGh0dHA6IDgwCiAgICBodHRwczogNDQzCiAgdHlwZTogTG9hZEJhbGFuY2VyCmNvbnRhaW5lclBvcnRzOgogIGh0dHA6IDgwODEKICBodHRwczogODQ0NAptYXJpYWRiOgogIGVuYWJsZWQ6IHRydWUKICBhdXRoOgogICAgZGF0YWJhc2U6IHRlc3Rfb3NjbGFzc19kYXRhYmFzZQogICAgdXNlcm5hbWU6IHRlc3Rfb3NjbGFzc191 ...
```
You have not updated the HTTPs port here (nor in the verify json pipeline), though the test is adapted.

mdhont replied:
I'm using the default port because it only works with that.

joancafom reviewed (changes requested):
Please, review my comments and re-test in order to check that https port `444` works as expected

joancafom reviewed (approved):
LGTM! Thanks for your PR

agomezmoron reviewed (approved):
LGTM, thanks!
